### PR TITLE
Exception best practices

### PR DESCRIPTION
**Pull request by AmmonRoberts**
Changed exception throwing to better follow best practices to preserve stack trace.

### CODE REVIEW

File: Snowflake.Data/Core/FileTransfer/SFFileTransferAgent.cs [hunk 1]
```diff
@@ -732,7 +732,8 @@ private ISFRemoteStorageClient renewExpiredClient()
             }
             catch (Exception ex)
             {
-                throw ex;
+                Console.WriteLine("Unhandled exception while uploading file.", ex);
```
Comment (Grauenwolf) [inline Snowflake.Data/Core/FileTransfer/SFFileTransferAgent.cs:735]: Shouldn't this be writing to `Logger`, not `Console`?

Comment (AmmonRoberts) [reply Snowflake.Data/Core/FileTransfer/SFFileTransferAgent.cs:735]: It most definitely should! Let me fix that real fast, thanks!

**Comment by Grauenwolf** [comment]
Let's see if they actually pick this up. Given how old some rather obvious PRs are, I'm tempted to to give up and create my own fork of the connector.

**Comment by alexrosenfeld10** [comment]
@Grauenwolf I have had private fork of this for quite some time. This repo doesn't get much attention. I've found the only / best way to get changes through if you really need them is to open a snowflake support case...

**Comment by Coder3333** [comment]
I just submitted a PR without realizing you had one along the same lines.  I took it further by fixing the ToString method of SnowflakeDbException, which currently hides the stack trace.  I really wish someone would take these PR's seriously; otherwise, why is this thing open source?

https://github.com/snowflakedb/snowflake-connector-net/pull/452

**Comment by Grauenwolf** [comment]
@alexrosenfeld10 @Coder3333 I created a public fork at https://github.com/TortugaResearch/Tortuga.Data.Snowflake/ 

It's a major refactor job, borderline a rewrite. But I think it puts the connector into a state where we can actually build upon it with confidence.

When you aren't busy, take a look at let me know what you think.

**Comment by sfc-gh-igarish** [comment]
@AmmonRoberts Changes look good. Do you mind to create an issue and then raise the PR? Sorry for late response. We are going through all PR  right now.

**Review by sfc-gh-igarish — APPROVED** [review]
LGTM

**Comment by sfc-gh-igarish** [comment]
You can merge it. Please create an issue for a PR. So we can track it.

**Pull request merged**